test/neural_networks/test_circuit_qnn.py

```diff
@@ -300,6 +328,90 @@ def test_circuit_qnn_gradient(self, config):
                 diff = weights_grad_ - grad
             self.assertAlmostEqual(np.max(np.abs(diff)), 0.0, places=3)
 
+    @data(
+        # sparse, sampling, quantum_instance_type, interpret (0=no, 1=1d, 2=2d), batch_size
+        (True, True, STATEVECTOR, 0, 1),
+        (True, True, STATEVECTOR, 0, 2),
+        (True, True, STATEVECTOR, 1, 1),
+        (True, True, STATEVECTOR, 1, 2),
+        (True, True, STATEVECTOR, 2, 1),
+        (True, True, STATEVECTOR, 2, 2),
+        (True, True, QASM, 0, 1),
+        (True, True, QASM, 0, 2),
+        (True, True, QASM, 1, 1),
+        (True, True, QASM, 1, 2),
+        (True, True, QASM, 2, 1),
+        (True, False, STATEVECTOR, 0, 1),
+        (True, False, STATEVECTOR, 0, 2),
+        (True, False, STATEVECTOR, 1, 1),
+        (True, False, STATEVECTOR, 1, 2),
+        (True, False, STATEVECTOR, 2, 1),
+        (True, False, STATEVECTOR, 2, 2),
+        (True, False, QASM, 0, 1),
+        (True, False, QASM, 0, 2),
+        (True, False, QASM, 1, 1),
+        (True, False, QASM, 1, 2),
+        (True, False, QASM, 2, 1),
+        (True, False, QASM, 2, 2),
+        (False, True, STATEVECTOR, 0, 1),
+        (False, True, STATEVECTOR, 0, 2),
+        (False, True, STATEVECTOR, 1, 1),
+        (False, True, STATEVECTOR, 1, 2),
+        (False, True, STATEVECTOR, 2, 1),
+        (False, True, STATEVECTOR, 2, 2),
+        (False, True, QASM, 0, 1),
+        (False, True, QASM, 0, 2),
+        (False, True, QASM, 1, 1),
+        (False, True, QASM, 1, 2),
+        (False, True, QASM, 2, 1),
+        (False, True, QASM, 2, 2),
+        (False, False, STATEVECTOR, 0, 1),
+        (False, False, STATEVECTOR, 0, 2),
+        (False, False, STATEVECTOR, 1, 1),
+        (False, False, STATEVECTOR, 1, 2),
+        (False, False, STATEVECTOR, 2, 1),
+        (False, False, STATEVECTOR, 2, 2),
+        (False, False, QASM, 0, 1),
+        (False, False, QASM, 0, 2),
+        (False, False, QASM, 1, 1),
+        (False, False, QASM, 1, 2),
+        (False, False, QASM, 2, 1),
+        (False, False, QASM, 2, 2),
```

Comment:
So many lines of data could be extracted to a separate file.

Reply:
Agree, but left as it is now. There should a consolidated change across all tests. And I'm not sure moving out such things to a separate file will solve the issue. Test parameters will be hidden, so I'm not convinced. Could be a dedicated function with a lot of nested loops better. But to be discussed, I think.

Reply:
Since it looks like the whole data set there for the test is easily computed, just combinations of the parameters, it seems like some sort of simplification should be easily doable in the future at some point if desired.